.github/workflows/ci.yaml

```diff
@@ -72,16 +70,115 @@ jobs:
       with:
         step: restore
-    - uses: actions/download-artifact@v2
+    - uses: actions/download-artifact@v3
       with:
         name: cryostat-core
         path: /home/runner/.m2/repository/io/cryostat/cryostat-core/
     - run: git submodule init
     - run: git submodule update --remote
-    - run: mvn -B -U clean verify
+    - run: mvn -B -U clean package
+    - name: Save cryostat image
+      run: podman save -o cryostat.tar --format oci-archive quay.io/cryostat/cryostat
+    - uses: actions/upload-artifact@v3
+      with:
+        name: cryostat
+        path: /home/runner/work/cryostat/cryostat/cryostat.tar
+    - uses: skjolber/maven-cache-github-action@v1
+      with:
+        step: save
+
+  integration-tests:
+    runs-on: ubuntu-latest
+    needs: [build-image]
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+        fetch-depth: 0
+    - uses: actions/setup-java@v2
+      with:
+        java-version: '17'
+        distribution: 'adopt'
+    - name: install xpath
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y libxml-xpath-perl
+    - uses: actions/download-artifact@v3
+      with:
+        name: cryostat
+    - name: Load cryostat image
+      run: podman load -i cryostat.tar
+    - uses: skjolber/maven-cache-github-action@v1
+      with:
+        step: restore
+    - run: bash repeated-integration-tests.bash
     - name: Print itest logs
       if: ${{ failure() }}
       run: ls -1dt target/cryostat-itest-*.log | head -n1 | xargs cat
+    - uses: skjolber/maven-cache-github-action@v1
+      with:
+        step: save
+
+  spotless:
+    runs-on: ubuntu-latest
+    needs: [build-deps]
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - uses: actions/setup-java@v2
+      with:
+        java-version: '17'
+        distribution: 'adopt'
+    - uses: skjolber/maven-cache-github-action@v1
+      with:
+        step: restore
+    - run: mvn spotless:check
+    - uses: skjolber/maven-cache-github-action@v1
+      with:
+        step: save
+
+  spotbugs:
+    runs-on: ubuntu-latest
+    needs: [build-deps]
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+        fetch-depth: 0
+    - uses: skjolber/maven-cache-github-action@v1
+      with:
+        step: restore
+    - uses: actions/setup-java@v2
+      with:
+        java-version: '17'
+        distribution: 'adopt'
+    - run: mvn compile spotbugs:check
+    - uses: skjolber/maven-cache-github-action@v1
+      with:
+        step: save
+
+  shellcheck:
+    runs-on: ubuntu-latest
+    needs: [build-deps]
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - uses: skjolber/maven-cache-github-action@v1
+      with:
+        step: restore
+    - run: mvn shellcheck:check
+    - uses: skjolber/maven-cache-github-action@v1
+      with:
+        step: save
+
+  build:
```

Comment:
Maybe this can be renamed `push-to-quay` or something.

It also still has a maven-cache save action occurring later down that looks like it can be removed with this new pared-down job definition.